File: 151003/Tolkanitsa/publisher/src/main/java/com/example/distributedcomputing/service/IssueService.java
package com.example.distributedcomputing.service;


import com.example.distributedcomputing.exceptions.NotFoundException;
import com.example.distributedcomputing.exceptions.AlreadyExist;
import com.example.distributedcomputing.mapper.IssueMapper;
import com.example.distributedcomputing.model.entity.Issue;
import com.example.distributedcomputing.model.request.IssueRequestTo;
import com.example.distributedcomputing.model.response.IssueResponseTo;
import com.example.distributedcomputing.repository.IssueRepository;
import jakarta.validation.Valid;
import jakarta.validation.constraints.Min;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.validation.annotation.Validated;

@Service
@RequiredArgsConstructor
@Validated
public class IssueService {
    private final CreatorService creatorService;
    private final IssueRepository repository;
    private final IssueMapper mapper;

    public Iterable<IssueResponseTo> getAll() {
        return mapper.entityToDto(repository.findAll());
    }
    public IssueResponseTo getById(@Min(0) Long id) {
        return repository.findById(id)
                .map(mapper::entityToDto)
                .orElseThrow();
    }
    public IssueResponseTo save(@Valid IssueRequestTo entity) {
        creatorService.getById(entity.creatorId());

        if (repository.findByTitle(entity.title()).isPresent()) {
            throw new AlreadyExist(403L, "Story with this title already exist");
        }
        return mapper.entityToDto(repository.save(mapper.dtoToEntity(entity)));
    }
    public IssueResponseTo update(@Valid IssueRequestTo entity) {
        Long id = entity.id();
        Issue editor = mapper.dtoToEntity(entity);
        editor.setId(id);
        return mapper.entityToDto(repository.save(editor));
    }
    public void delete(@Min(0) Long id) {
        repository.findById(id).orElseThrow(() -> new NotFoundException(404L, "Not founded"));
        repository.deleteById(id);
    }
}

File: 151004/Afanasenko/appLab1/src/main/kotlin/com/github/kornet_by/dc/lab1/service/IssueService.kt
package com.github.kornet_by.dc.lab1.service

import com.github.kornet_by.dc.lab1.dto.request.IssueRequestTo
import com.github.kornet_by.dc.lab1.dto.request.IssueRequestToId
import com.github.kornet_by.dc.lab1.dto.response.IssueResponseTo

interface IssueService {
	suspend fun create(requestTo: IssueRequestTo?): IssueResponseTo?

	suspend fun deleteById(id: Long): Boolean

	suspend fun getAll(): List<IssueResponseTo>

	suspend fun getById(id: Long): IssueResponseTo?

	suspend fun update(requestTo: IssueRequestToId?): IssueResponseTo?
}